Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29584 Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h**
* #29536 Use c10::variant-based enums for SmoothL1Loss module and functional
* #29535 Use c10::variant-based enums for F::grid_sample
* #29404 Make all non-input arguments to functionals part of its options, and pass functional options by const-ref when possible

In Python, `float` dtype is always 64-bit (https://stackoverflow.com/a/8216110), and the C++ equivalent APIs should take `double` dtype to match the bit length.

Differential Revision: [D18436616](https://our.internmc.facebook.com/intern/diff/D18436616)

Commit:
Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h

kostmo commented:
## CircleCI build failures summary
As of commit 4b157c0:
* **1/1** failures introduced in this PR
* **0/1** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=4b157c0bcf421e9f45398b4a71f0db996d29cb8b).

---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

Commit:
Update on "Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h"

Change arg dtype from float to double in LPPool and nn/utils/clip_grad.h

gh-metadata: pytorch pytorch 29584 gh/yf225/62/head
